### PR TITLE
Internal mock node clean up

### PR DESCRIPTION
**Pull request by shamsasari**
* InMemoryMessagingNetwork.InMemoryMessaging renamed to MockNodeMessagingService and moved to internal package
* start method added to MockNodeMessagingService which enables AbstractNode to call makeMessagingService in its c'tor
* Removed TopicStringValidator as it's no longer used
* Clean up of TestStartedNode
* Merged InMemoryMessagingTests into InternalMockNetworkTests as it's testing InternalMockNetwork

### CODE REVIEW

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/MockNodeMessagingService.kt [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+                                               override val senderSeqNo: Long? = null,
+                                               /** Note this flag is never set in the in memory network. */
+                                               override val isSessionInit: Boolean = false) : ReceivedMessage {
+
+        override val additionalHeaders: Map<String, String> = emptyMap()
+    }
+
+    private inner class InMemoryDeduplicationHandler(override val receivedMessage: ReceivedMessage, val transfer: InMemoryMessagingNetwork.MessageTransfer) : DeduplicationHandler, ExternalEvent.ExternalMessageEvent {
+        override val externalCause: ExternalEvent
+            get() = this
```
Comment (distributedleetravis) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/MockNodeMessagingService.kt]: This probably doesn't need to be a getter - just assign `this` to the field?

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/MockNodeMessagingService.kt [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+        override val deduplicationHandler: DeduplicationHandler
+            get() = this
```
Comment (distributedleetravis) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/MockNodeMessagingService.kt]: Ditto

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/InternalMockNetwork.kt [hunk 3]
```diff
@@ -308,25 +318,11 @@ open class InternalMockNetwork(defaultParameters: MockNetworkParameters = MockNe
...
-        }
-
         val mockNet = args.network
         val id = args.id
+        init {
+            require(id >= 0) { "Node ID must be zero or positive, was passed: $id" }
+        }
         private val entropyRoot = args.entropyRoot
         var counter = entropyRoot
         override val log get() = staticLog
```
Comment (distributedleetravis) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/InternalMockNetwork.kt]: Does this need to be a getter?

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/InternalMockNetwork.kt [hunk 4]
```diff
@@ -343,9 +339,23 @@ open class InternalMockNetwork(defaultParameters: MockNetworkParameters = MockNe
 
         override val started: TestStartedNode? get() = uncheckedCast(super.started)
```
Comment (distributedleetravis) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/InternalMockNetwork.kt]: Is the cast needed? (and if not, is the override needed?) `super` is `AbstractNode<TestStartedNode>`, so I think its `started` already has the type `TestStartedNode?`

**Comment by shamsasari** [comment]
I'm OK with the getters; they each represent one less field.

**Pull request merged**